pypfopt/efficient_frontier/efficient_frontier.py

```diff
@@ -337,16 +340,21 @@ def efficient_risk(self, target_volatility, market_neutral=False):
                 )
             )
 
-        self._objective = objective_functions.portfolio_return(
-            self._w, self.expected_returns
-        )
-        variance = objective_functions.portfolio_variance(self._w, self.cov_matrix)
+        update_existing_parameter = self.is_parameter_defined('target_variance')
+        if update_existing_parameter:
+            self.update_parameter_value('target_variance', target_volatility ** 2)
+        else:
+            self._objective = objective_functions.portfolio_return(
+                self._w, self.expected_returns
+            )
+            variance = objective_functions.portfolio_variance(self._w, self.cov_matrix)
 
-        for obj in self._additional_objectives:
-            self._objective += obj
+            for obj in self._additional_objectives:
+                self._objective += obj
 
-        self._constraints.append(variance <= target_volatility ** 2)
-        self._make_weight_sum_constraint(market_neutral)
+            target_variance = cp.Parameter(name="target_variance", value=target_volatility ** 2)
+            self.add_constraint(lambda _: variance <= target_variance)
```

Comment:
In this case, there is no need for a lambda function, but I verified that the result is still correct. We could also allow adding constraints that contain `_w` without the lambda callable, but I am not sure if that would be better.